stable/kube-state-metrics/templates/service.yaml

```diff
@@ -12,11 +12,18 @@ metadata:
     prometheus.io/scrape: '{{ .Values.prometheusScrape }}'
   {{- end }}
 spec:
+  type: {{ .Values.service.type }}
   ports:
   - name: "http"
     protocol: TCP
     port: {{ .Values.service.port }}
+    {{- if .Values.service.nodePort }}
```

Comment:
Nit: `if` and `end` statements should be indented one level less than the lines they wrap.